src/algorithms/machinelearning/tensorflowpredict.cpp

```diff
@@ -293,8 +311,28 @@ TF_Output TensorflowPredict::graphOperationByName(const char* nodeName,
   TF_Output output = {TF_GraphOperationByName(_graph, nodeName), index};
 
   if (output.oper == nullptr) {
-    throw EssentiaException("TensorflowPredict: Can't init node names.");
+    vector<string> nodes = TensorflowPredict::nodeNames();
+
+    string exceptionText = "TensorflowPredict: '" + string(nodeName) + "' is not a valid node name of this graph.\n" +
+                           "Available node names are:\n";
+
+    for (vector<string>::const_iterator i = nodes.begin(); i != nodes.end() - 1; ++i) exceptionText +=  *i + ", ";
```

Comment:
Looks like this code doubles E_INFO output by the `configure()` method

Reply:
When the first element of `inputs` is an empty string the list of available nodes is printed through E_INFO, but when it is an invalid node name (not existing in the graph) this list is printed as part of the exception message.
It's true that there are a couple of duplicated lines, but this permits to have the method `nodeNames` returning a vector<string> which is more flexible than returning the list of nodes as a single string and it may be useful in the future.